packages/common-test-utils/src/index.ts

```diff
@@ -810,3 +810,104 @@ export class NoteTestPresetsV2 {
     ];
   }
 }
+
+/**
+ * Utility class to make the tests against vault to be concise while having descriptive
+ * failure messages. */
+export class VaultAsserter {
+  private readonly workspaceVault: WorkspaceVault;
+
+  constructor(workspaceVault: WorkspaceVault) {
+    this.workspaceVault = workspaceVault;
+  }
+
+  /** Asserts that no match is found  */
+  async noMatch(noMatchList: string[]) {
+    const hasNoneMatched = await EngineTestUtilsV4.checkVault({
+      ...this.workspaceVault,
+      nomatch: noMatchList,
+    });
+
+    assert(
```

Comment:
Plugin tests are written in `mocha` to work with VSCode. We are using `jest` to test our plugin, so we wrap assertions to be jest-like in `testUtils`. We should keep things consistent and switch these into jest like `expect`s.

Reply:
jest's expects do not seem to allow custom error messages to be added to the calls, which prompted the creation of these wrapper classes. Given lack of messages, and that when the tests fail, they give line numbers in JS output instead of TS source code. Adding error messages adds quite a bit of clarity of what is failing. 

There are way to add messages to jest, but they require custom matchers or adding additional dependency hence thought just using assert here would be a simpler approach, since its hidden from the user of tests anyway and will throw an error similar how jest will throw an error. The main difference being of JestAssertionError being thrown vs AssertionError being thrown.

Reply:
As per later discussion refactoring to use the current way.